### ISSUE TITLE
Sentry.LoggerBackend doesn't take context set by `Sentry.Context.set_extra_context` into account

### ISSUE THREAD

**Issue by syfgkjasdkn** [getsentry/sentry-elixir#491]
### Environment

How do you use Sentry? `self-hosted, but it doesn't matter`, error is on the client's side

Which SDK and version? `sentry-elixir 8.0.6`

### Steps to Reproduce

1. Use logger backend with `Sentry.Context.set_extra_context`
2. Crash
3. No `additional info` form `Sentry.Context.set_extra_context` in the sentry issue

### Expected Result

`:extra` is not lost

### Actual Result

`:extra` is lost

**Comment by syfgkjasdkn** [comment]
it happens because sentry logger backend sets `:extra` key twice,

1. https://github.com/getsentry/sentry-elixir/blob/a77b709bdee83d2b923ee56ad39821e70002b6ea/lib/sentry/logger_backend.ex#L104
2. https://github.com/getsentry/sentry-elixir/blob/a77b709bdee83d2b923ee56ad39821e70002b6ea/lib/sentry/logger_backend.ex#L106

Which results in `opts` for `capture_message` like this:

```elixir
stacktrace: [
   # ...
  ],
  event_source: :logger,
  level: "error",
  # set by logger
  extra: %{logger_level: :error, logger_metadata: %{}},
  result: :none,
  extra: %{
    # set by Sentry.Context.set_extra_context
  },
  tags: %{
    # set by Sentry.Context.set_extra_context
  },
  user: %{
    # set by Sentry.Context.set_extra_context
  }
```

**Comment by syfgkjasdkn** [comment]
The fix is to merge `sentry[:extra]` with ` %{logger_metadata: logger_metadata(meta, state), logger_level: level}`

**Comment by mitchellhenke** [comment]
@syfgkjasdkn yes, a PR would be great!

**Comment by ccapndave** [comment]
I also ran into this problem!  I guess @sl0thentr0py is busy so I will make a PR :+1:

**Comment by lubien** [comment]
Just ran into this issue, would be lovely to see this fix

**Comment by nickdichev-firework** [comment]
This fix isn't on 8.0.6, right? Are there plans to cut a hex release sometime soon?
